implementations/rust/ockam/ockam_transport_smoltcp/src/net/device/packet_pool.rs

```diff
@@ -0,0 +1,77 @@
+// This is copied verbatim(Some stuff we don't need removed) from [embassy](https://github.com/embassy-rs/embassy/blob/7561fa19348530ce85e2645e0be8801b9b2bbe13/embassy-net/src/packet_pool.rs)
+// One advantage of using atomic_pool instead of heapless it doesn't seem to have the same kind of ABA problem.
```

Comment:
The **main** reason to use `atomic_pool` was because `embassy` was doing that.

I'm not sure the advantages of `heapless`'s Treiber stack over `atomic_pool`'s  method of keeping track of available pool using a bitset.

While the advantage of `atomic_pool`'s method is clear, it is `Sync` for all architectures, I'm not sure if it's because `atomic_pool` didn't take the same considerations and edge cases into account or the implementation is sound. 

Also, `atomic_pool` doesn't mention the ABA soundness in x64 problem and by superficially reviewing it I came to the conclusion that it doesn't seem to have that problem since that seems to arise from the Treiber stack using a linked list but I can't confirm it %100.

I will read more on this and get back to you! (Atomics are really complicated 😵‍💫)